go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/stackrox/rox
 
-go 1.21
+go 1.21.8
```

Comment:
It's not clear to me why this happens. `go get` outputs that it's performing this change when upgrading ClairCore, so my best guess is due to ClairCore having a minimum version of Go 1.21.8:

```
❯ go get github.com/quay/claircore@v1.5.27
go: upgraded go 1.21 => 1.21.8
go: upgraded github.com/google/pprof v0.0.0-20231023181126-ff6d637d2a7b => v0.0.0-20240409012703-83162a5b38cd
go: upgraded github.com/quay/claircore v1.5.25 => v1.5.27
go: upgraded go.opentelemetry.io/otel v1.24.0 => v1.25.0
go: upgraded go.opentelemetry.io/otel/metric v1.24.0 => v1.25.0
go: upgraded go.opentelemetry.io/otel/trace v1.24.0 => v1.25.0
go: upgraded modernc.org/libc v1.41.0 => v1.49.3
go: upgraded modernc.org/memory v1.7.2 => v1.8.0
go: upgraded modernc.org/sqlite v1.29.2 => v1.29.8
```

Reply:
You probably have `GOTOOLCHAIN=auto` (see `go env GOTOOLCHAIN`). That's the default for the upstream `go` version, and it proved to be a pain for me when trying to backport fixes into older StackRox versions which used go1.20.

I set mine to `GOTOOLCHAIN=path`:

https://go.dev/doc/toolchain
```
When GOTOOLCHAIN is set to <name>+auto or <name>+path (or the shorthands auto or path), the go command selects and runs a newer Go version as needed. Specifically, it consults the toolchain and go lines in the current workspace’s go.work file or, when there is no workspace, the main module’s go.mod file. If the go.work or go.mod file has a toolchain <tname> line and <tname> is newer than the default Go toolchain, then the go command runs <tname> instead. If the file has a toolchain default line, then the go command runs the default Go toolchain, disabling any attempt at updating beyond <name>. Otherwise, if the file has a go <version> line and <version> is newer than the default Go toolchain, then the go command runs go<version> instead.
```

If you use that, I don't believe it'll keep messing with the `go.mod`

Reply:
Tested it, and the `go` version was updated to `go 1.21.8`. Now that I think about it, it makes sense. This defines the minimum version required to build this codebase, and ClairCore requires 1.21.8 as the minimum. Therefore, we now require 1.21.8 as the minimum.

Reply:
Writing it here because I keep forgetting:

`go` directive = minimum Go version required to use the module

`toolchain` directive = suggested toolchain to use (cannot be less than `go` directive). This only affects things if it's the main module (so if ClairCore set a `toolchain`, it'd not affect us. Only their `go` directive does)

Reply:
https://go.dev/ref/mod

Reply:
> This defines the minimum version required to build this codebase, and ClairCore requires 1.21.8 as the minimum. Therefore, we now require 1.21.8 as the minimum.

This ^ was my understanding after trying to grok `toolchain` yet again yesterday afternoon.